pkg/odo/cli/application/delete.go

```diff
@@ -56,7 +60,12 @@ func (o *DeleteOptions) Run() (err error) {
 		return err
 	}
 
-	if o.force || ui.Proceed(fmt.Sprintf("Are you sure you want to delete the application: %v from project: %v", o.appName, o.Project)) {
+	if o.OutputFlag == "json" {
```

Comment:
In this case, please take care of the `err = printDeleteAppInfo(o.Client, o.appName, o.Project)` above. The output looks weird, is printed in the terminal and does not resemble json
```
[mrinaldas@localhost odo]$ odo app delete -o json
This application has following components that will be deleted
 component named  nodejs-3-5-nodejs-qpvv
```

Reply:
Added patch, now it won’t show the app info when machinereadable flag is  present